libraroo-backend/src/shared/models/query-options.ts

```diff
@@ -1,8 +1,7 @@
 import { FilterQuery } from 'mongoose';
+import { PaginationDto } from './pagination.dto';
 
-export class QueryOptions<T> {
-  maxResults: number = 10;
-  startIndex: number = 0;
+export class QueryOptions<T> extends PaginationDto{
   sort?: { [key: string]: 1 | -1 };
   filter?: FilterQuery<T>;
 }
```

Comment:
I dont think this change is correct, in PaginationDto the properties are nullable. 
Either make them non nullable with default values so they can be used in QueryOptions (make sure when they arent submited to the API, the default values are used).
Or just revert the change to QueryOptions, im fine with this code being duplicated cause these serve differant perposes (exposing pagination API, and interacting with the DB)